docarray/array/annlite.py

```diff
@@ -7,3 +7,13 @@
 class DocumentArrayAnnlite(StorageMixins, DocumentArray):
     def __new__(cls, *args, **kwargs):
         return super().__new__(cls)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args, **kwargs):
+        """
+        Ensures that offset2ids are stored in the db after
+        operations in the DocumentArray are performed.
+        """
+        self._save_offset2ids()
```

Comment:
1. promote this to one of the base class, so that you don't need to repeatedly writing it
2. unify the name of `_save_offset2ids`, `_update_offset2ids_meta()` 
3. by doing 1 and 2, all storage backend should not need to implement them

Reply:
true, refactored